### PR TITLE
feat: fox farming old contracts

### LINKED ISSUE #2468: User should be able to withdraw from old FOX farming Contracts
### Overview

So, in looking at mobile wallet migration, we realized that there will be some users who are still staked in the old FOX farming contracts on the legacy mobile app. Therefore, we'll need some functionality in the web app for those users to get their tokens out of the old contracts.

Those users should be able to:
1. See that they have LP tokens staked in an inactive farm (contracts 1,2 and 3)
2. See the positions on the FOX and ETH asset pages and in the defi overview section
3. Be able to Unstake from those contracts (this should only allow them to withdraw EVERYTHING and claim their staking rewards).
5. The APR for all those contracts should be displayed as 0% and there should be a label on the card in the overview that says "Ended"

Those users should NOT be able to:
1. Stake any more into those old contracts (no more staking is allowed). The stake button should hidden.
2. Claim separately. The only allowed action is withdraw max, while also claiming.
3. Edit the withdrawal amount. The MAX withdrawal amount should be displayed and not editable.

Notes:
There are some UI impacts to the modal, to accomplish this (See mock ups below).

### References and additional details

Here is a list of the old FOX Farming contracts that will need to be displayed, should a user have funds in any of them:

Round 3: https://etherscan.io/address/0x212ebf9fd3c10f371557b08e993eaab385c3932b
Round 2: https://etherscan.io/address/0xc54B9F82C1c54E9D4d274d633c7523f2299c42A0
Round 1: https://etherscan.io/address/0xDd80E21669A664Bce83E3AD9a0d74f8Dad5D9E72

### Acceptance Criteria

The overview pretty much outlines the acceptance criteria

### Need By Date

Sept 1

### Screenshots/Mockups
There are UI impacts as seen here. 

**Screen 1:**
1. Stake and claim options are hidden
2. Withdraw & Claim should be the only button 
3. If the user clicks the "withdrawal available" on the rewards row, just take them directly into the withdrawal screen.

**Screen 2.**
1. The % buttons are not visible
2. The withdrawal amount displays the entire balance, and the "Rewards you'll receive" should appear below it.
3. The withdrawal amount is not editable.
Screen Shot 2022-08-18 at 9 25 33 AM
Screen Shot 2022-08-18 at 9 25 48 AM

### Estimated effort

_No response_

### PR DESCRIPTION
**Pull request by stackedq**
## Description

this PR adds support for expired v1-3 fox contracts.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #2468

## Risk

none

## Testing

1. enable the `FoxFarming` flag2. 
2. if you have a balance in an expired contract, skip to step 4. if not deposit into fox farming v4 ( you need to make sure `ConsiderFoxFarmingV4ExpiredForTesting` feature flag is off to be able to deposit into v4.) 
3. enable  the `ConsiderFoxFarmingV4ExpiredForTesting` feature flag
4. open up an expired opportunity
5. you should just have the permission to withdraw and claim
6. for re-testing you can use steps 2 & 3 instructions.7. 
 

## Screenshots (if applicable)
Screen Shot 2022-08-19 at 21 50 04
Screen Shot 2022-08-19 at 21 55 50
Screen Shot 2022-08-19 at 21 48 04
Screen Shot 2022-08-19 at 21 48 12

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -50,6 +50,7 @@ const validators = {
   REACT_APP_FEATURE_MULTI_CURRENCY: bool({ default: false }),
   REACT_APP_FEATURE_FOX_LP: bool({ default: false }),
   REACT_APP_FEATURE_FOX_FARMING: bool({ default: false }),
+  REACT_APP_FEATURE_CONSIDER_FOX_FARMING_V4_EXPIRED_FOR_TESTING: bool({ default: false }),
```
Comment (gomesalexandre) [inline src/config.ts:53]: Missing from `sample.env`

File: src/context/FoxEthProvider/FoxEthProvider.tsx [hunk 2]
```diff
@@ -157,24 +183,34 @@ export const FoxEthProvider = ({ children }: FoxEthProviderProps) => {
...
-            expired,
-            apy: lpApr ? bnOrZero(apr).plus(lpApr).toString() : '0',
+            expired: isOpportunityExpired,
+            apy: lpApr && !isOpportunityExpired ? bnOrZero(apr).plus(lpApr).toString() : '0',
             tvl,
           }
         }),
       )
-      const totalOpBalances = newOpportunities.reduce(
+      // show opportunities that are not expired or expired but user have balance in them
```
Comment (gomesalexandre) [inline src/context/FoxEthProvider/FoxEthProvider.tsx:203]: ?
```suggestion
      // show opportunities that are not expired or ended but user have balance in them
```

File: src/context/FoxEthProvider/FoxEthProvider.tsx [hunk 3]
```diff
@@ -157,24 +183,34 @@ export const FoxEthProvider = ({ children }: FoxEthProviderProps) => {
...
+      const visibleOpportunities = newOpportunities.filter(
+        opportunity =>
+          !opportunity.expired || (opportunity.expired && bnOrZero(opportunity.cryptoAmount).gt(0)),
+      )
+      const totalOpBalances = visibleOpportunities.reduce(
```
Comment (gomesalexandre) [inline src/context/FoxEthProvider/FoxEthProvider.tsx:208]: ser
```suggestion
      const totalOpportunitiesBalances = visibleOpportunities.reduce(
```

**Review by gomesalexandre — COMMENTED** [review]
First pass, LGTM overall - we will need @shapeshift/product to provide us and/or test with a wallet they mentioned had some amount in the old farming contracts, but from testing with the monkey-ish patch provided in this PR it does what it says on the box.

Also noticed that after successfully claiming and withdrawing, the ended opportunity is not considered active anymore and thus, rugs us at confirm step with an infinite spinner:

image

**Comment by stackedq** [comment]
colors changed per Product request.
The "you have a withdrawal available" box is link to withdraw modal now.

Screen Shot 2022-08-23 at 08 48 48
Screen Shot 2022-08-23 at 08 48 58
Screen Shot 2022-08-23 at 08 49 06

**Review by gomesalexandre — APPROVED** [review]
Stamping as previously conceptually approved and was waiting on product testing, who gave their green light on this, and all occurences of the temporary `REACT_APP_FEATURE_CONSIDER_FOX_FARMING_V4_EXPIRED_FOR_TESTING` are now removed.

**Pull request merged**